.github/workflows/ghci.yml

```diff
@@ -0,0 +1,146 @@
+name: GitHub CI tests
+
+on:
+  push:
+    branches: [master, ci, coverity_scan]
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, macos-10.15]
+        cc: [gcc, clang]
+        exclude:
+          - os: macos-10.15
+            cc: gcc
+        include:
+          # Enable distcheck for one build
+          - os: ubuntu-18.04
+            cc: gcc
+            do_distc: yes
+          # Run Coverity on a clang build; Coverity's gcc causes issues
+          - os: ubuntu-18.04
+            cc: clang
+            do_coverity: yes
+
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+
+      - name: retrieve caches
+        uses: actions/cache@v2
+        with:
+          path: |
+            $HOME/Library/Caches/Homebrew
+            /usr/local/Homebrew
+          key: ${{ runner.os }}-${{ matrix.os }}-build
+
+      - name: install deps
+        run: |
+          if (echo "${{ matrix.os }}" | grep -qE '^macos-') ; then
+            #brew update
+            # freetype, autoconf and libtool are preinstalled
+            # and `brew install` fails if a non-uptodate version is already installed
+            #brew upgrade freetype autoconf libtool
+            brew install automake fribidi harfbuzz nasm
+          else
+            #sudo apt-get update && sudo apt-get upgrade
```

Comment:
Are these commented `update` and `upgrade` invocations kept in case we at some point require a dependency newer than GitHub’s default version?

Reply:
Yes. In case of `brew` it also lists the used preinstalled dependencies which will be handy should we actually need update & upgrade. The preinstalled-distinction is not necessary with `apt`, its comment is just there to match the `brew` clause.

Reply:
Apparently GitHub does not keep their images up-to-date enough to allow us to skip `apt update`. (see: [this build failure](https://github.com/wangqr/Aegisub/runs/1665185971?check_suite_focus=true#step:3:447))

I guess this means we should also be calling`brew update`, but then *(MacOS builöds are significantly slower and)* [this](https://github.com/TheOneric/Aegisub/runs/1674614692?check_suite_focus=true#step:3:4318) brew-link-error appears again...
For now I've enabled `update` for `apt` only.

Reply:
Other projects like [TS-Aegisub](https://github.com/TypesettingTools/Aegisub/blob/417ba4f/.github/workflows/ci.yml#L83-L88) and  [hugo](https://github.com/gohugoio/hugo/blob/master/.github/workflows/test.yml#L41-L43) omit `brew update` in their GHA-scripts as well; so I guess it's probably fine.